src/replacer.js

```diff
@@ -78,28 +78,33 @@ export default class Replacer {
 			do {
 				prevContent = content;
 				if (from) {
-					let simpleTo = !this.literal && !this.replace;
+					let simpleTo = typeof to !== "function" && !this.literal && !this.replace;
 					if (to !== undefined || this.replace) {
 						content = content.replaceAll(from, simpleTo ? to : (...args) => {
 							let resolvedArgs = resolveReplacementArgs(args);
-							to = this.to ?? this.insert ?? resolvedArgs.match;
+							let ret = resolvedArgs.match; // no change if no to
+							to = this.to ?? this.insert;
 
-							if (!this.literal && to !== undefined) {
-								// Replace special replacement patterns
-								to = emulateStringReplacement(resolvedArgs, to);
+							if (to !== undefined) {
+								if (typeof to === "function") {
+									ret = to.call(this, ...args);
```

Comment:
Here, we should probably pass either `resolvedArgs` here, or `ret` (since it's a match)

Reply:
Or both?

Reply:
Yes, I'm in two minds on whether it should be consistent with the native JS signature for replace callbacks or the much better one afforded via `resolvedArgs`…

Reply:
Probably the latter, but let's get this working first

Reply:
Passing `ret` make it work (from my experiments)

Reply:
Oh I see what you mean

Reply:
Now?

Reply:
It works! Yay! 🥳

Reply:
It's orthogonal, but I would probably add `with` to the list of exceptions. Or should it be capitalized, too?

Reply:
Yay! Can you add the tests to this PR? And something to the docs?

Not sure about with. It seemed weird to me to capitalize it, but I think per the style guides it should be. But that's secondary.

Reply:
I added the test. Not sure if I'm able to write anything at all right now. Can try to do it in the morning.